### ISSUE TITLE
TypeError: insertText() got an unexpected keyword argument 'idx'

### ISSUE THREAD

**Issue by cerha** [pymupdf/PyMuPDF#265]
Hello,

in versions > 1.14.2 I get the following error when calling insertPage():

```
>> import fitz
>> doc = fitz.open()
>> doc.insertPage(text="Some text")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../python2.7/site-packages/fitz/fitz.py", line 2158, in insertPage
    color = color, idx = idx, set_simple = set_simple)
TypeError: insertText() got an unexpected keyword argument 'idx'
```

Kindest regards

Tomáš Cerha

**Comment by JorjMcKie** [comment]
Thanks for catching this!
I will remove ``idx`` parameter in the next patch.
You know how to get around this problem, right?
Use `page=doc.newPage()` and then ``page.insertText()``

**Comment by JorjMcKie** [comment]
fixed in v1.14.11

**Comment by cerha** [comment]
Perfect.  Thank you!

**Comment by cerha** [comment]
Sorry, not perfect yet.  Same code, different error.  Seems unrelated so I opened a new issue #269.

**Comment by HaraldLieder** [comment]
Having massive internet issues at the moment, so putting an e-mail on stack in the hope it will get transmitted some time …

Located the error – sorry about that.
Circumvention: please specify “doc.insertPage(…, fontname=’helv’, …)”.

This will also become the default (should already have been).
This method is rarely used and I am considering to deprecate it. It should be replaced with two separate statements anyway:

page = doc.newPage(…)
page.insertText(…)

So you will to wait for the next patch unfortunately …

Jorj

Von: Tomáš Cerha<mailto:notifications@github.com>
Gesendet: Montag, 18. März 2019 19:09
An: pymupdf/PyMuPDF<mailto:PyMuPDF@noreply.github.com>
Cc: Subscribed<mailto:subscribed@noreply.github.com>
Betreff: Re: [pymupdf/PyMuPDF] TypeError: insertText() got an unexpected keyword argument 'idx' (#265)


Sorry, not perfect yet. Same code, different error. Seems unrelated so I opened a new issue #269<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpymupdf%2FPyMuPDF%2Fissues%2F269&data=02%7C01%7C%7C2b0ddbe950044e5997e408d6abf6b66e%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636885473435836753&sdata=KYPWaGNCsZ33y6iFjtjuDWShOZfLk24AHpv8b%2B3WQsk%3D&reserved=0>.

**Comment by cerha** [comment]
No problem for me.  Just reported to let you know.  Thank you for cooperation!  Tomas.

**Comment by JorjMcKie** [comment]
Thank you - and thanks for reporting this.
Took the chance to clean the code in that area. `insertPage` was old code, and ironically `newPage` was coded on top of it.
Now it is the other way round, and `insertPage` follows my own recommendation: (1) newPage, (2) page.insertText ... as it should be.

**Comment by JorjMcKie** [comment]
fixed with v1.14.11, v1.14.12
